sentry-android-core/src/main/java/io/sentry/android/core/SendCachedEnvelopeIntegration.java

```diff
@@ -29,7 +39,8 @@ public SendCachedEnvelopeIntegration(
   @Override
   public void register(@NotNull IHub hub, @NotNull SentryOptions options) {
     Objects.requireNonNull(hub, "Hub is required");
-    final SentryAndroidOptions androidOptions =
+    this.hub = hub;
```

Comment:
you can merge the assignment
```suggestion
    this.hub = Objects.requireNonNull(hub, "Hub is required");
```